src/main/webapp/app/quiz/re-evaluate/quiz-re-evaluate-warning.controller.js

```diff
@@ -61,41 +61,96 @@
                         backUpQuestion = backUpQuiz.questions[i];
                     }
                 }
-                if (backUpQuestion != null) {
+                if (backUpQuestion !== null) {
                     // question set invalid?
                     if (question.invalid !== backUpQuestion.invalid) {
                         vm.questionInvalid = true;
                     }
-                    // answer deleted?
-                    if (question.answerOptions.length !== backUpQuestion.answerOptions.length) {
-                        vm.answerDeleted = true;
-                    }
                     // question scoring changed?
                     if (question.scoringType !== backUpQuestion.scoringType) {
                         vm.scoringChanged = true;
                     }
-                    //check each answer
-                    question.answerOptions.forEach(function (answer) {
-                        // only check if there are no changes on the answers yet
-                        if (!vm.answerCorrectess || !vm.answerInvalid) {
-                            var backUpAnswer = null;
-                            for (var j = 0; backUpAnswer === null && j < backUpQuestion.answerOptions.length; j++) {
-                                if (backUpQuestion.answerOptions[j].id === answer.id) {
-                                    backUpAnswer = backUpQuestion.answerOptions[j];
+                    //check MultipleChoiceQuestions
+                    if (question.type === "multiple-choice") {
+                        // question-Element deleted?
+                        if (question.answerOptions.length !== backUpQuestion.answerOptions.length) {
+                            vm.questionElementDeleted = true;
+                        }
+
+                        //check each answer
+                        question.answerOptions.forEach(function (answer) {
+                            // only check if there are no changes on the question-elements yet
+                            if (!vm.questionCorrectness || !vm.questionElementInvalid) {
+                                var backUpAnswer = null;
+                                for (var j = 0; backUpAnswer === null && j < backUpQuestion.answerOptions.length; j++) {
```

Comment:
can you please use a for each loop here?

Reply:
@MoritzIssig It seems you are trying to find an element in the array here. Take a look at the `find()` function in javascript arrays: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
(Don't worry about IE compatibility, I just added es6-shim in the improvePerformance branch)